src/nodes/types.ts

```diff
@@ -13,7 +13,7 @@ type NodeAddress = {
 };
 
 type NodeMapping = {
-  [key: string]: NodeAddress;
+  [key: NodeId]: NodeAddress;
 };
```

Comment:
This `NodeMapping` is only used for seed nodes. The name of the type suggests more generic usage.

Reply:
If it's only used for the seed nodes then it could use a more specific name. However the type itself is pretty generic and could be used elsewhere if needed.

Reply:
Ok but it's also incorrect in its type. The `NodeId` is not a valid type in POJO keys. So I've changed it to:

```ts
type NodeMapping = Record<NodeIdString, NodeAddress>;
```

Reply:
However I don't like how `NodeMapping` is assuming that it is `NodeIdString`.

This `NodeMapping` is likely to be user-configurable, and later in the future we would put it in our `config.ts`.

Very likely we would configure it to use a fully encoded ID, not the binary string because it comes from the outside world.

I would prefer this be called `type SeedNodes = Record<NodeIdEncoded, NodeAddress>;` instead.

This is one of the few exceptions to use the encoded versions instead of the binary string from since it is exposed to the outside world.

Reply:
This has been changed over, as well as changes to `src/bin/utils/parsers.ts`, tests and how the NCM processes them.